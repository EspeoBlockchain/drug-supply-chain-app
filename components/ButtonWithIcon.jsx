import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  Image,
} from 'react-native';
import PropTypes from 'prop-types';
import colors from '../constants/Colors';
import fonts from '../constants/Fonts';

const styles = StyleSheet.create({
  button: {
    marginVertical: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    height: 135,
  },
  buttonText: {
    color: colors.white,
    fontSize: 24,
    fontFamily: fonts.main,
    flexWrap: 'wrap',
    flex: 1,
  },
  register: {
    backgroundColor: colors.registerLight,
  },
  registerText: {
    color: colors.registerDark,
  },
  transport: {
    backgroundColor: colors.transportLight,
  },
  transportText: {
    color: colors.transportDark,
  },
  verify: {
    backgroundColor: colors.verifyLight,
  },
  verifyText: {
    color: colors.verifyDark,
  },
  icon: {
    width: 50,
    height: 50,
  },
});

const buttonWithIcon = (props) => {
  const {
    onPress,
    buttonStyle,
    buttonTextStyle,
    text,
    img,
  } = props;

  return (
    <TouchableOpacity onPress={onPress}>
      <View style={[styles.button, styles[buttonStyle]]}>
        <Text style={[styles.buttonText, styles[buttonTextStyle]]}>{text}</Text>
        <Image source={img} style={styles.icon} />
      </View>
    </TouchableOpacity>
  );
};

buttonWithIcon.propTypes = {
  onPress: PropTypes.func,
  buttonStyle: PropTypes.string,
  buttonTextStyle: PropTypes.string,
  text: PropTypes.string,
  img: PropTypes.number,
};

buttonWithIcon.defaultProps = {
  onPress: () => null,
  buttonStyle: '',
  buttonTextStyle: '',
  text: '',
  img: 0,
};

export default buttonWithIcon;
