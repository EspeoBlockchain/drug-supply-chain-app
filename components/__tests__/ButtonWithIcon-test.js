import 'react-native';
import React from 'react';
import ButtonWithIcon from '../ButtonWithIcon';
import renderer from 'react-test-renderer';

it('renders corectly', () => {
    const tree = renderer.create(<ButtonWithIcon />).toJSON();

    expect(tree).toMatchSnapshot();
});