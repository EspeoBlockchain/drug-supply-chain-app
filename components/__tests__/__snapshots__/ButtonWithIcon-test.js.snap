// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corectly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 135,
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingHorizontal": 20,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "flex": 1,
            "flexWrap": "wrap",
            "fontFamily": "raleway",
            "fontSize": 24,
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Image
      source={0}
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
